sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -0,0 +1,209 @@
+            colStats += attr.name -> BasicColStats(
+              dataType = attr.dataType,
+              numNulls = rowCount - statsRow.getLong(pos + StringStatsAgg.numNotNullsIndex),
+              maxColLen = Some(statsRow.getLong(pos + StringStatsAgg.maxLenIndex)),
+              avgColLen =
+                Some(statsRow.getLong(pos + StringStatsAgg.sumLenIndex) / (1.0 * rowCount)),
+              ndv = Some(statsRow.getLong(pos + StringStatsAgg.ndvIndex)))
+            pos += StringStatsAgg.statsSeq.length
+          case BinaryType =>
+            colStats += attr.name -> BasicColStats(
+              dataType = attr.dataType,
+              numNulls = rowCount - statsRow.getLong(pos + BinaryStatsAgg.numNotNullsIndex),
+              maxColLen = Some(statsRow.getLong(pos + BinaryStatsAgg.maxLenIndex)),
+              avgColLen =
+                Some(statsRow.getLong(pos + BinaryStatsAgg.sumLenIndex) / (1.0 * rowCount)))
+            pos += BinaryStatsAgg.statsSeq.length
+          case BooleanType =>
+            val numOfNotNulls = statsRow.getLong(pos + BooleanStatsAgg.numNotNullsIndex)
+            val numOfTrues = Some(statsRow.getLong(pos + BooleanStatsAgg.numTruesIndex))
+            colStats += attr.name -> BasicColStats(
+              dataType = attr.dataType,
+              numNulls = rowCount - numOfNotNulls,
+              numTrues = numOfTrues,
+              numFalses = numOfTrues.map(i => numOfNotNulls - i),
+              ndv = Some(2))
+            pos += BooleanStatsAgg.statsSeq.length
+        }
+      }
+      newStats = newStats.copy(basicColStats = colStats.toMap)
+      sessionState.catalog.alterTable(catalogTable.copy(stats = Some(newStats)))
+      // Refresh the cached data source table in the catalog.
+      sessionState.catalog.refreshTable(tableIdent)
+    }
+
+    Seq.empty[Row]
+  }
+
+  private def statsAgg(name: String, dataType: DataType): Seq[Column] = dataType match {
+    // Currently we only support stats generation for atomic types
+    case n: NumericType => NumericStatsAgg(name)
+    case TimestampType | DateType => NumericStatsAgg(name)
+    case StringType => StringStatsAgg(name)
+    case BinaryType => BinaryStatsAgg(name)
+    case BooleanType => BooleanStatsAgg(name)
+    case otherType =>
+      throw new AnalysisException(s"Analyzing column $name of $otherType is not supported.")
+  }
+}
+
+object ColumnStats extends Enumeration {
+  val MAX, MIN, NDV, NUM_NOT_NULLS, MAX_LENGTH, SUM_LENGTH, NUM_TRUES = Value
+}
+
+trait StatsAggFunc {
+  // This sequence is used to track the order of stats results when collecting.
+  val statsSeq: Seq[ColumnStats.Value]
+
+  def apply(name: String): Seq[Column] = {
+    val col = Column(name)
+    statsSeq.map {
+      case ColumnStats.MAX => max(col)
+      case ColumnStats.MIN => min(col)
+      // count(distinct col) will have a shuffle, so we use an approximate ndv for efficiency
+      case ColumnStats.NDV => approxCountDistinct(col)
```

Comment:
The error rate of the estimation should be tunable, right? See Hive has such a configuration: `hive.stats.ndv.error`